reference/shaders-msl/comp/barriers.comp

```diff
@@ -0,0 +1,79 @@
+#pragma clang diagnostic ignored "-Wmissing-prototypes"
```

Comment:
@billhollings I included this, but this output seems wrong to me. memoryBarrierBuffer() is wrong here. Looking at the implementation in spirv_msl.cpp it doesn't seem to take into account that OpControlBarrier has its own memory barriers as well (typically Workgroup).